designdocs/logical_batching.md

```diff
@@ -0,0 +1,67 @@
+# Summary
+
+In other documents, we explore ways to structure an explicit batch of messages.
+That is, in terms of concrete structure. This document has a different aim --
+deriving the idea of a batch logically, as a product of fields present on other
+entities in our system.
+
+## Background
+
+We are currently experimenting with a new system architecture that has the
+following entities:
+
+1. Schema
+   - Id
+   - Format
+   - FormatType
+   - PayloadLocation
+2. Message
+   - Off-chain / On-chain Payload
+
+*Protocols* would be responsible for indicating not only the shape of the
+underlying payload, but also its format and location. This is important, because
+a protocol's format property would indicate whether a given payload contains a
+batch or not.
+
+## FormatType
+
+Payloads can either be of format type `PARQUET` or `AVRO`. These file formats
+indicate specifically whether their payloads are singular or arrays of objects
+(More info needed here).
+
+## PayloadLocation
+
+The combination of format and location constrain possible payload types:
+
+```txt
+| Format  | Location        | Payload           |
+-------------------------------------------------
+| Avro   | On-chain         | Public Graph      |
+| Parquet| On-chain         | ??                |
+| Avro   | IPFS (Off-chain) | Private Graph (?) |
+| Parquet| IPFS (Off-chain) | Announcement      |
+```
```

Comment:
The payload column should just be Examples. Perhaps something like:
```suggestion
```txt
| Format  | Location        | Examples           |
-------------------------------------------------
| Avro   | On-chain         | DSNP Graph Change      |
| Parquet| On-chain         | Unknown                |
| Avro   | IPFS (Off-chain) | Larger Avro structures |
| Parquet| IPFS (Off-chain) | DSNP Broadcast or Reply Announcements      |
```
```